### ISSUE TITLE
How can I prevent CPU overflow while optimizing tensorflow with particle swarm optimization?

### ISSUE THREAD

**Issue by furkanatlan** [tensorflow/tensorflow#53544]
I am trying to optimize the following hyperparameters with pso for segmentation in modified Unet model.

Hyperparameters: Dropout, learning_rate, factor, min_lr

tensorflow==2.0.0

keras==2.3.1

PSO iterations number = 10

PSO Particle number = 10

10 models are running in each iteration (PSO iteration). This means a total of 100 models will run (model_1, model_2,...,model_100). But, when I go to the 2nd iteration, I get a memory overflow error on the 19th model. For the solution of this, I looked and applied similar problems on this platform, but the error did not go away. How can I avoid this memory overflow? I also tried the following code to solve this problem (for tensorflow 1.* version compatibility). First Soution Try:

```
import tensorflow as tf
from tensorflow.python.keras import backend as K
import os
os.environ["CUDA_VISIBLE_DEVICES"] = "-1"
config = tf.compat.v1.ConfigProto(intra_op_parallelism_threads=36,inter_op_parallelism_threads=2, allow_soft_placement=True, device_count = {'CPU': 36})
sess = tf.compat.v1.Session(config=config)
K.set_session(tf.compat.v1.Session(config=config))
```

THE CODE (Current State): The code block below is run on the calculation set with 36 Cores requested in the slurm script file.

```
def objective_function(x):
    _ = tf.Variable([1])
    context._context = None
    context._create_context()
    tf.config.threading.set_intra_op_parallelism_threads(36)
    tf.config.threading.set_inter_op_parallelism_threads(2)
    Dropoutt = x[0]
    Llearning_rate = x[1]
    Factorr = x[2]
    Mmin_lr = x[3]
    with tf.device('/CPU:0'):
        model = get_unet_mod(input_img, n_filters=16, dropout=Dropoutt, batchnorm=True)
        model_isim = model.name
        model.compile(optimizer=Adam(amsgrad=True, learning_rate=Llearning_rate), loss=jaccard_distance_loss, metrics=["accuracy", dice_coef, f1])
        Callbackss = [ReduceLROnPlateau(monitor="loss",mode="min",factor=Factorr, patience=10, min_lr=Mmin_lr, verbose=0)]
        results = model.fit(X_train, y_train, batch_size=8, verbose=0, epochs=30, callbacks=Callbackss, validation_data=(X_test, y_test))

PSO(objective_function, bounds, particle_size, iterations)
```
Error:

slurmstepd: error: step 8207716.4294967294 hit memory+swap limit at least once during execution. this may or may not result in some failure.

**Comment by tilakrayal** [comment]
@furkanatlan ,
We see that you are using tf version 2.0,  please update to latest stable  v2.7 and let us know if you are using same issue.Thanks!

**Comment by furkanatlan** [comment]
The issueis not related to the version of tensorflow. The issue is that
when optimizing keras model.fit with PSO, it requests more memory than
allocated and the code doesn't run due to memory overflow. I would be very
happy if you could help me with this.
Best regards

tilakrayal ***@***.***>, 27 Ara 2021 Pzt, 06:51 tarihinde
şunu yazdı:

**Comment by tilakrayal** [comment]
@furkanatlan ,
I ran the code shared and face a different error, please find the gist [hehttps://colab.research.google.com/gist/tilakrayal/55f75a374aea8d3a0fd21a7a3c8ee7f3/untitled167.ipynbre](url).Please share all dependencies to replicate the issue or share a colab gist with the reported error.

**Comment by tilakrayal** [comment]
I ran the code shared and face a different error, please find the gist [here](https://colab.research.google.com/gist/tilakrayal/55f75a374aea8d3a0fd21a7a3c8ee7f3/untitled167.ipynb).Please share all dependencies to replicate the issue or share a colab gist with the reported error.

**Comment by tilakrayal** [comment]
Its unlikely for TF 2.0 version to receive any bug fixes except when we have security patches. There is a high possibility that this was fixed with later TF versions. Perhaps you can use latest tf versions for your case. Thanks!
